torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -351,13 +352,15 @@ def call_function(
 
         # branches
         assert isinstance(
-            args[1], (UserFunctionVariable, NestedUserFunctionVariable)
+            args[1],
+            (UserFunctionVariable, NestedUserFunctionVariable, NNModuleVariable),
```

Comment:
If you want, splitting out the tuple change into a separate PR would make it easier to review

Reply:
Sounds good!